### PR TITLE
Issue 29 - added linear and quadratic model

### PR DESCRIPTION
**Pull request by klaferr**
- Created a LinearModel and QuadModel (within linear_model.py) which is used as a base function in the related lag and accumulation models. 
- Updated the __init__.py file to load in both methods from the new linear_model.py file
- test_LinearTroughs notebook shows that both new models technically function.

### CODE REVIEW

File: mars_troughs/linear_model.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    linearCoeff: float = 1e-6,
+    quadCoeff: float = 1e-6,
+
+    @property
+    def parameters_names(self) -> Dict[str, float]:
+        return['intercept', 'linearCoeff', 'quadCoeff']
+                 
+    def eval(self, x):
+        p = [self.intercept, self.linearCoeff, self.quadCoeff]
+        return sum((a*x**i for i,a in enumerate(p)))
```
Comment (tmcclintock) [inline mars_troughs/linear_model.py:36]: This line seems troubling. If `x` is a numpy array of length `N`, wouldn't calling `sum()` collapse this array into a single number?

**Comment by tmcclintock** [comment]
This is a good start! There are a few tweaks we should make before merging this though:

- since each of the linear models (e.g. `LinearLag`) inherit the `slope` and `intercept` parameters from the `LinearModel` mixin class, we can deprecate the identically-named parameters from those child classes (for instance [this line](https://github.com/Bramson-Planetary-Research-Group/Mars-Troughs/blob/cc8f0f3d630c7dcb2dca3435380e0d1dfedb1240/mars_troughs/lag_model.py#L70))
- let's remove the commented-out lines of code

File: mars_troughs/accumulation_model.py [hunk 2]
```diff
@@ -109,7 +110,8 @@ def get_accumulation_at_t(self, time: np.ndarray) -> np.ndarray:
             accumulation rates A, in m/year
 
         """
-        return self.intercept + self.slope * self._ins_data_spline(time)
+        return self.eval(time)
```
Comment (tmcclintock) [inline mars_troughs/accumulation_model.py:113]: @klaferr is it this line that's busted?

Comment (klaferr) [reply mars_troughs/accumulation_model.py:113]: No the line that causes the problem is the return line in get_yt, I get an error saying that there is no self.intercept if I remove the init call with that variable included.

Comment (tmcclintock) [reply mars_troughs/accumulation_model.py:113]: Ok shoot, I know why this is broken. I'll write up the solution now. The TLDR is that we are using "multiple inheritance" and the `__init__` function of the second parent class (`LagModel`) is not being called.

Comment (tmcclintock) [reply mars_troughs/accumulation_model.py:113]: I retract what I wrote before -- MI is not the problem. Can you clarify what tests fail? I pulled this branch and commented out lines 95 and 96, and all the tests still passed.

File: test/test_accumulation_model.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+from mars_troughs import (
+    LinearInsolationAccumulation,
+    QuadraticInsolationAccumulation,
+)
+
+
+class LinearAccumulationTest(TestCase):
+    def test_smoke(self):
+        # this is testing if classes functions as expected, so
+        # insolation and time don't need to be real - right?
```
Comment (tmcclintock) [inline test/test_accumulation_model.py:14]: Correct.

**Comment by tmcclintock** [comment]
This looks great! Nice job @klaferr. @kristelizq I think you should take a quick look at this PR and then we are good to merge.

**Pull request merged**